server/index.js

```diff
@@ -4,116 +4,147 @@ const app = express();
+        console.log(err);
+      })
+  }
 });
 
 //returns price and promos for an array of product ids
-app.get('/PriceAndPromotion/multiple/:product_ids', (req, res) => {
-  let arrayOfIds = JSON.parse(req.params.product_ids);
-  let dataBundle = async () => {
-    let priceAndPromosArray = []
-    for (let i = 0; i < arrayOfIds.length; i++) {
-      await PriceAndPromo.find({ product_id: arrayOfIds[i] })
-        .then((doc) => {
-          let data = {
-            product_id: doc[0].product_id,
-            price: doc[0].price,
-            promotion: doc[0].discount
-          }
-          priceAndPromosArray.push(data);
-        })
-        .catch(err => console.log(err))
+app.get('/PriceAndPromotion/multiple/:product_ids', async (req, res) => {
+
+  try {
+    if(!req.params.product_ids){
+      res.status(400).send('No Product Id');
+    } else if (req.params.product_ids) {
+      
+      let arrayOfIds = JSON.parse(req.params.product_ids);
+          
+      let arrayOfRecords = await getPriceAndPromotion(arrayOfIds);
+
+      let arrayOfPriceandPromtions = await arrayOfRecords.map( record => {
+
+        let {id, base_price, discount, max} = record;
+        let product_id = id;
+        let price = base_price;
+        let promotion = base_price * (discount/100);
+        promotion = Number(promotion.toFixed(2));
+        
+        if(max) {
+          price *= (1 - max/100)
+          price = Number(price.toFixed(2));
+        }
+
+        return {product_id, price, promotion};
+
+      });
+    
+      res.status(200).send(arrayOfPriceandPromtions);
     }
-    return priceAndPromosArray;
+    
+  } catch (err) {
+    res.status(404).send('Data Not Found');
+    console.log(err);
   }
-  Promise.resolve(dataBundle()).then((data) => {
-    res.status(200).send(data);
-  })
+
 });
 
-/**     Extend CRUD Operations      **/
+
 
 //based on product number from URL,returns an object that includes price, promotion/discount, start date, and expiry
 app.get('/:product_id', (req, res) => {
   res.sendFile(path.resolve('public/index.html'));
 });
 
-//creates a new product
+/**     Extend CRUD Operations      **/
+
+//inserts one record
 app.post('/PriceAndPromotion', (req, res) => {
 
-  let priceAndPromotionData = req.body;
+  let data = req.body;
 
-  //needs to have all fields to post record
-  if(!priceAndPromotionData.product_id || !priceAndPromotionData.price || !priceAndPromotionData.start || !priceAndPromotionData.expiry) {
-    res.status(400).send('Missing Field(s)');
-  } else {
-    PriceAndPromo.create(priceAndPromotionData)
-    .then( () => {
-      res.status(201).send('Post Successful');
-    })
-    .catch( (err) => {
-      res.status(404).send(err);
-    });
-  }
+  insertRecords(data)
+    .then( (response) => res.status(201).send({"NumberOfInsertedRecords": response.length, "ProductIds": response}))
+    .catch( (err) => res.status(400).send({error: JSON.stringify(err)}));
```

Comment:
Technically, a catch would take care of error handling, whether it's an incorrect post body or a db error, but maybe differentiating between the two with more granular error handling (code 400's vs 500's) would result in a better client experience / clarity? Just a thought!